Add pull request template to get more context when PR's are open

FabioRosado opened the pull request:
I've copied the template that we are using on the main opsdroid repo and decided to use it on the site as well since we had a PR that got merged but didn't close the issue since it didn't have the text closes/fixes <issue number>

Hopefully this template will motivate contributors to explain why they decided that a change should be made when opening a PR.

FabioRosado committed:
Add pull request template to get more context when PR's are open

SolarDrew commented:
Are the type of change and checklist sections still relevant for the website? On board with adding the template though, good call.

FabioRosado committed:
Remove checklist and type of change

FabioRosado commented:
Good point, I was wondering if I should delete those bits or not haha 
Just updated the file and deleted them since you are right, they don't seem too relevant for the site 😄

SolarDrew commented:
👍 LGTM
